packages/language/src/parser/parser.ts

```diff
@@ -2238,88 +2237,77 @@ export class PliParser extends AbstractParser {
         CstNodeKind.DefineOrdinalStatement_CloseParenValues,
       );
     });
-    this.OPTION1(() => {
+
+    // zero or more attributes following
+    this.MANY(() => {
+      // either signed/unsigned OR a precision attribute
       this.OR2([
         {
+          // signed | unsigned
           ALT: () => {
-            this.CONSUME_ASSIGN1(tokens.SIGNED, (token) => {
+            this.OR3([
```

Comment:
Suggestion: Please don't nest `OR` calls, unless absolutely necessary. This alternative needs to be evaluated multiple times in order to actually be parsed.

Reply:
Good point, I can flatten this out under a single alternative.